wordsFrequency/main.cpp

```diff
@@ -0,0 +1,129 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <utility>
+#include <vector>
+#include <cctype>
+#include <list>
+#include <algorithm>
+#include <limits>
+#include <optional>
+#include <set>
+#include <cassert>
+
+void addToFirstTen(std::vector<std::pair<std::string, int>>& words, const std::string& word, int frequency) {
```

Comment:
Non mi piace granchè `std::pair`, trovo che sia una classe estremamente generica e che renda il codice meno leggibile.
In questi casi preferisco sempre una piccola `struct`, magari in stile [POD](https://stackoverflow.com/questions/146452/what-are-pod-types-in-c), con un nome sensato a livello di classe e di membri.

Vedere sparso in giro per la base di codice un sacco di `std::pair<string, int>` mi mette un po' in crisi, perchè non capisco se quella coppia stringa-intero faccia sempre riferimento allo stesso tipo di dato aggregato oppure no.